tests/phpunit/class-test-core-sitemaps.php

```diff
@@ -159,5 +159,31 @@ public function test_core_sitemaps_xml() {
 		$this->assertSame( $expected, $xml, 'Sitemap page markup incorrect.' );
 	}
 
+	/**
+	 * Test robots.txt output.
+	 */
+	public function test_robots_text() {
+		// Get the text added to the default robots text output.
+		$robots_text = apply_filters( 'robots_txt', '', true );
+		$sitemap_string = 'Sitemap: http://' . WP_TESTS_DOMAIN . '/?sitemap=index';
+
+		$this->assertTrue( false !== strpos( $robots_text, $sitemap_string ), 'Sitemap URL not included in robots text.' );
+	}
+
+	/**
+	 * Test robots.txt output with permalinks set.
+	 */
+	public function test_robots_text_with_permalinks() {
+		// Set permalinks for testing.
+		$this->set_permalink_structure( '/%year%/%postname%/' );
+
+		// Get the text added to the default robots text output.
+		$robots_text = apply_filters( 'robots_txt', '', true );
+		$sitemap_string = 'Sitemap: http://' . WP_TESTS_DOMAIN . '/sitemap.xml';
 
+		$this->assertTrue( false !== strpos( $robots_text, $sitemap_string ), 'Sitemap URL not included in robots text.' );
+
+		// Clean up permalinks.
+		$this->set_permalink_structure();
```

Comment:
Should run this before the assertion. Otherwise this code is never reached when the assertion fails.